packages/core/src/client.test.ts

```diff
@@ -1992,6 +2009,101 @@ describe('Client', () => {
     }
   });
 
+  describe('maxRetries', () => {
+    test('should try 3 times by default', async () => {
+      const fetch = mockFetch(500, serverError(new Error('Something is broken')));
+      const client = new MedplumClient({ fetch });
+
+      await expect(client.get(client.fhirUrl('Patient', '123'))).rejects.toThrow(
+        'Internal server error (Error: Something is broken)'
+      );
+      expect(fetch).toHaveBeenCalledTimes(3);
+    });
+
+    test('should only try once when maxRetries = 0', async () => {
+      const fetch = mockFetch(500, serverError(new Error('Something is broken')));
+      const client = new MedplumClient({ fetch });
+
+      await expect(client.get(client.fhirUrl('Patient', '123'), { maxRetries: 0 })).rejects.toThrow(
+        'Internal server error (Error: Something is broken)'
+      );
+      expect(fetch).toHaveBeenCalledTimes(1);
+    });
+
+    test.each([400, 401, 404])('%d status code is not retried', async (statusCode) => {
```

Comment:
What about 429 — should that get retried, or nah?

Reply:
That's a good callout... probably not...
We might want to include that one

Reply:
The 429 response should include how much time until they're allowed to make the request.  Ideal behavior would be to read that header, and sleep accordingly.  That could be future PR though.

Reply:
We would probably want some more intelligent logic than just sleeping based on time given, ie. if the time is really long we probably don't want to wait for it in most cases